### DISCUSSION TITLE
Consider to decrease major version releases for year

### DISCUSSION THREAD

**Discussion by ABuffEr** [nvaccess/nvda#12682] [General]
Hi,
I have no idea how much it could be simple or not, so this is only a consideration.
2021.1 stable is quite new today, and it has been released after the half of the year (precisely on July 12), with add-on API upgrade that we all know (and that users could afraid even if it's not present).
The last year, on July 28, was releasing 2020.2.
I say it without any criticism, I follow all discussions and issues on this repo, so I see the huge amount of work that a small organization like NVAccess has to address. The effort is extremely valuable.
2021.2 and 2021.3, I imagine, could arrive for the second half and end of the year, respectively. And this sounds reasonable.
But, like 2020.4, also 2021.4 could be ready for first months of 2022. And this sounds like a version born old.
So, to sumarize... I have the impression that, for devs and users, 2 or 3 releases in a year could be better, bringing, on the long time, not only to a better reputation of releases, but also to versions with more features for users and a minor amount of (even small) compatibility refactoring for add-on developers (more changes, but less often).
Any thoughts on this?

**Comment by seanbudd** [comment]
I'm a little confused about the conclusion drawn here. I understand that the labelling of releases seems a bit strange as they have drifted later and later, but I'm not sure why that means the frequency should decrease. If anything, a faster cadence should make it easier to determine whether to label/target a release for the next year or the current year. I would err on the side of a faster release schedule, so that important bug fixes reach users ASAP, and support for newer software (like Windows 11) doesn't lag behind far. 

Code refactor is sadly a necessary part of the code - any feature driven development is tied to eliminating earlier technical debt. The ratio of new features to refactor is something that is hard to improve over the long term, with diminishing returns. It depends on the current technical debt you have and the changes you need to make. Stricter code review helps minimise technical debt being introduced in the first place, making future development faster. The ratio of features to refactor is unusual in 2021.1's case, as it was a comprehensive removal of old deprecated code, changes that were flagged for removal almost a decade ago in some cases.

**Comment by Brian1Gaff** [comment]
Well considering that the next Python upgrade is going to completely break 
anyone using Windows 7, and there are still a lot, it might with the 
consideration of windows 11 coming along almost out of the blue and the bug 
in the new version of Python, maybe slowing this move down to med next year, 
as it would leave time for the Python folk to make sure the various bits of 
python work in the new OS. We are also to see the new virtual windows 
running in the cloud for corporate very soon now and that might well be an 
issue for NVDA as well.

 I think as a free option for many we do need to be mindful of the fact that 
there are especially now far more people still running old windows. Of 
course I know that nvda can rightly say that  if you must run older windows, 
older frozen versions of nvda exist for that purpose, but it also might be 
wise to protect an old add on place too for those people.
 It may of course be that Windows 11 is the best thing since sliced bread, 
in which case, onwards and upwards, but if nvda wants to be used by 
corporate the new Windows over the web systems need to be researched for 
possible issues.

Just some thoughts.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message ----- 
From: "Alberto Buffolino" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Sunday, July 25, 2021 11:12 AM
Subject: [nvaccess/nvda] Consider to decrease major version releases in this 
year (#12682)
